packages/eslint-plugin/src/rules/no-unused-vars.ts

```diff
@@ -408,10 +408,12 @@ export default util.createRule<Options, MessageIds>({
         return cached;
       }
 
-      for (const statement of node.body?.body ?? []) {
-        if (statement.type === AST_NODE_TYPES.TSExportAssignment) {
-          MODULE_DECL_CACHE.set(node, true);
-          return true;
+      if (node.body && node.body.type !== AST_NODE_TYPES.TSModuleDeclaration) {
```

Comment:
in this context body will never be a `TSModuleDeclaration` because path selector excludes it